app/src/main/java/com/sun/moviedb/view/childhome/ChildHomeViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package com.sun.moviedb.view.childhome
+
+import android.app.Application
+import androidx.lifecycle.*
+import com.sun.moviedb.base.BaseResponse
+import com.sun.moviedb.data.dto.CategoryDto
+import com.sun.moviedb.data.entity.Movie
+import com.sun.moviedb.data.repository.MovieByCategoryRepository
+
+/**
+ * Created by nguyenxuanhoi on 2019-08-15.
+ * @author nguyen.xuan.hoi@sun-asterisk.com
+ */
+class ChildHomeViewModel(application: Application, val reponsitory: MovieByCategoryRepository) :
+    AndroidViewModel(application) {
+
+    companion object {
+        private val FIRST_PAGE = 1
+    }
+
+    fun getMoviesByCategory(categories: String): LiveData<BaseResponse<List<Movie>>> =
+        reponsitory.getMoviesByCategory(categories, FIRST_PAGE)
+
+    val categories = reponsitory.getQueryTypeCategory()
```

Comment:
https://developer.android.com/topic/libraries/architecture/viewmodel#implement
em thử tham khảo xem google viết có hay k nhé

Reply:
Anh cho em hỏi ở hàm trên getMoviesByCategory em truyền vào queryType. Sau đó bên ChildHomeFragment em for categories rồi gọi tới getMoviesByCategory sau đó observe nó nhận dữ liệu là List<Movie>. submit vào CategoryDTO. 
Làm như vậy nó sẽ ko dữ được dữ liệu khi thay đổi config. Vây có viết nào khác không ạ.
Em thử mấy cách những toàn bị lặp code =(